iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergSplit.java

```diff
@@ -70,14 +78,31 @@ public String[] getLocations() {
     // getLocations() won't be accurate when called on worker nodes and will always return "*"
     if (locations == null && conf != null) {
       boolean localityPreferred = conf.getBoolean(InputFormatConfig.LOCALITY, false);
-      locations = localityPreferred ? Util.blockLocations(task, conf) : ANYWHERE;
+      locations = localityPreferred ? blockLocations(task, conf) : ANYWHERE;
     } else {
       locations = ANYWHERE;
     }
 
     return locations;
   }
 
+  private static String[] blockLocations(ScanTaskGroup<FileScanTask> task, Configuration conf) {
```

Comment:
due to the type change, we can't use Util.blockLocations? we'll probably need to raise the iceberg request to avoid code duplication

Reply:
I created a PR https://github.com/apache/iceberg/pull/11053, and add a comment
https://github.com/apache/hive/pull/5409/commits/4206c61a80c06db7a8f899bb848ae65091eb1857